src/reference/asciidoc/amqp.adoc

```diff
@@ -361,6 +361,19 @@ The `ConnectionFactory` argument can be used to distinguish target connection na
 By default a `beanName` of the `AbstractConnectionFactory` and an internal counter are used to generate `connection_name`.
 The `<rabbit:connection-factory>` namespace component is also supplied with the `connection-name-strategy` attribute.
 
+
+The connection might be blocked for interaction from the Broker according to the https://www.rabbitmq.com/memory.html[Memory Alarm].
+Starting with _version 2.0_, the `org.springframework.amqp.rabbit.connection.Connection` can be supplied with the `com.rabbitmq.client.BlockedListener` s to to be notified for connection block and unblock events.
+In addition the `AbstractConnectionFactory` emits a `ConnectionBlockedEvent` and `ConnectionUnblockedEvent`, respectively, via its internal `BlockedListener` implementation.
+These can be useful in application logic to react appropriately to the problems on the Broker and perform some maintenance actions for example.
+
+IMPORTANT: When single `CachingConnectionFactory` is used in the application, e.g. in case of auto-configuration by Spring Boot, it is important to understand that by default it provides a single shared connection instance.
```

Comment:
~~IMPORTANT: When single `CachingConnectionFactory` is used in the application, e.g. in case of auto-configuration by Spring Boot, it is important to understand that by default it provides a single shared connection instance.~~
IMPORTANT: When the application is configured with a single `CachingConnectionFactory` ...  ,as is the default with Spring Boot auto-configuration,  the application will stop working when the connection is blocked by the broker.